Src/BootCamp.Chapter/Lesson4.cs

```diff
@@ -0,0 +1,171 @@
+            return response.StartsWith("Y") || response.StartsWith("y");
+        }
+
+        public static void PromptUserForStats()
+        {
+
+            string name = PromptNameInput("What is your name?");
+            int age = PromptAgeInput("How old are you? If I may ask");
+            float weight = PromptFloatInput("What is your weight? In Kilograms please.");
+            float height = PromptFloatInput("How tall are you? In Meters please.");
+
+            bool isHeightAndWeightValid = ValidateHeightAndWeight(weight, height);
+
+            if (isHeightAndWeightValid)
+            {
+                float calculatedBMI = GetBmi(weight, height);
+                Console.WriteLine($"{name} is {age} years old, weight is {weight} kg with a height of {height} meters.");
+                Console.WriteLine($"The calculated BMI  is: {calculatedBMI:F}");
+            }
+            else
+            {
+                Console.WriteLine("The values you provided were not valid.");
+            }
+
+        }
+
+        // Validation Functions
+        public static string PromptNameInput(string promptMessage)
+        {
+            Console.WriteLine(promptMessage);
+
+            var name = Console.ReadLine();
+            if (string.IsNullOrWhiteSpace(name))
+            {
+                Console.Write("Name cannot be empty.");
+                return "-";
+            }
+            else
+            {
+                return name;
+            }
+        }
+        
+        public static int PromptAgeInput(string promptMessage)
+        {
+            Console.WriteLine(promptMessage);
+            var age = Console.ReadLine();
+            if (IsNumber(age) && IsPositive(float.Parse(age)))
+            {
+                return int.Parse(age);             
+            }
+            else if (string.IsNullOrEmpty(age))
+            {
+                return 0;
+            }
+            else
+            {
+                DisplayInvalidInputErrorMessage(age);
+                return -1;
+            }
+        }
+
+        public static float PromptFloatInput(string promptMessage)
+        {
+            Console.WriteLine(promptMessage);
+
+            var value = Console.ReadLine();
+            if (IsNumber(value) && IsPositive(float.Parse(value)))
+            {
+                return float.Parse(value);
+            }
+            else if(string.IsNullOrEmpty(value))
+            {
+                return 0;
+            }
+            else
+            {
+                DisplayInvalidInputErrorMessage(value);
+                return -1;
+            }
+        }
+       
+        public static bool ValidateHeightAndWeight(float weight, float height)
+        {
+            bool isValidWeight = IsPositive(weight);
+            bool isValidHeight = IsPositive(height);
+            return ((isValidHeight && isValidWeight) && (weight > height));
+        }
+
+        public static float GetBmi(float weight, float height)
```

Comment:
It's not GetBmi that you do. It sounds like you just get it from some properties. What you actually do is calculate Bmi.

Reply:
Since the actual Bmi calculation below, you should be careful with naming. Maybe this function does 2 things and thus should be split into two?
ValidateBmi and CalculateBmi